test/uk/gov/hmrc/cipphonenumberverification/services/AuditServiceSpec.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cipphonenumberverification.services
+
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{times, verify, when}
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.{ArgumentCaptor, IdiomaticMockito}
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.libs.json.{Json, OWrites, Writes}
+import play.api.test.Helpers.POST
+import play.api.test.{FakeHeaders, FakeRequest}
+import uk.gov.hmrc.cipphonenumberverification.audit.VerificationRequestAuditEvent
+import uk.gov.hmrc.cipphonenumberverification.models.AuditType
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditResult.{Failure, Success}
+import uk.gov.hmrc.play.audit.http.connector.{AuditConnector, AuditResult}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class AuditServiceSpec extends AnyWordSpec with Matchers with IdiomaticMockito {
+
+  // NOTE - SOLVED 5 MATCHERS EXPECTED, 2 FOUND ERROR MESSAGE.
+  // FIX IS DON'T PRESS RETURN/NO LINE SPACE ON MOCKITO LINES
+  "sendEvent" should {
+    "send AuditEvent to audit service and audit service call is a successful result" in new SetUp {
+      val auditResultSuccess: AuditResult = Success
+      val auditEvent = VerificationRequestAuditEvent("myPhoneNumber", "myPasscode")
+      when(mockAuditConnector.sendExplicitAudit[VerificationRequestAuditEvent](any[String], any[VerificationRequestAuditEvent])(any[HeaderCarrier], any[ExecutionContext], any[Writes[VerificationRequestAuditEvent]]))
```

Comment:
Can we use the IdiomaticMockito syntax here. See other spec classes for examples

Reply:
In the case of the audit service class, I can't get it to compile.  Please can you find an example.
i.e. A scala equivalent of thenAnswer

Reply:
There are quite a few examples for example, in NotificationServiceSpec:

      mockGovUkConnector.notificationStatus(notificationId)
        .returns(Future.successful(Right(httpResponse)))

Reply:
In this case, it would return Future.unit